lib/carrierwave/storage/fog.rb

```diff
@@ -1,6 +1,10 @@
 # encoding: utf-8
 
-require "fog" unless defined?(::Fog)
+begin
+  require "fog/core"
+rescue LoadError
+  require "fog" unless defined?(::Fog)
```

Comment:
Here there was no `rescue LoadError` because if fog isn't available and this file is required, it should raise.

Your `rescue LoadError` is necessary here because we want to load all of fog if it wasn't already loaded by the `require "fog/core"`.